domain-services/fiscal-event-post-processor/src/main/java/org/egov/service/FiscalEventDereferenceEnrichmentService.java

```diff
@@ -2,14 +2,43 @@
 
 
 import lombok.extern.slf4j.Slf4j;
+import org.egov.common.contract.AuditDetails;
+import org.egov.common.contract.request.RequestHeader;
+import org.egov.util.FiscalEventPostProcessorUtil;
+import org.egov.web.models.FiscalEvent;
 import org.egov.web.models.FiscalEventDeReferenced;
+import org.egov.web.models.FiscalEventRequest;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
+import java.util.UUID;
+
 @Service
 @Slf4j
 public class FiscalEventDereferenceEnrichmentService {
 
+    @Autowired
+    private FiscalEventPostProcessorUtil processorUtil;
+
+    public void enrich(FiscalEventRequest fiscalEventRequest, FiscalEventDeReferenced fiscalEventDeReferenced) {
+        RequestHeader requestHeader = fiscalEventRequest.getRequestHeader();
+        FiscalEvent fiscalEvent = fiscalEventRequest.getFiscalEvent();
+
+        fiscalEventDeReferenced.setEventTime(fiscalEvent.getEventTime());
+        fiscalEventDeReferenced.setEventType(fiscalEvent.getEventType().name());
+        fiscalEventDeReferenced.setParentReferenceId(fiscalEvent.getParentReferenceId());
+        fiscalEventDeReferenced.setParentEventId(fiscalEvent.getParentEventId());
+        fiscalEventDeReferenced.setReferenceId(fiscalEvent.getReferenceId());
+        fiscalEventDeReferenced.setAttributes(fiscalEvent.getAttributes());
+        fiscalEventDeReferenced.setVersion("1.0.0");
 
-    public void enrich(FiscalEventDeReferenced fiscalEventDeReferenced, FiscalEventDeReferenced fiscalEventDeReferenced1) {
+        AuditDetails auditDetails = null;
```

Comment:
AuditDetails are already present in the consumed record from the fiscal-event-service. Why do we need it here?

Reply:
yes. That make sense. will remove